packages/esm-service-queues-app/src/hooks/useQueue.ts

```diff
@@ -0,0 +1,14 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import useSWRImmutable from 'swr/immutable';
+import { type QueueServiceInfo } from '../types';
+import { useQueues } from './useQueues';
+
+export function useQueue(queueUuid?: string) {
+  const { queues, isLoading, error } = useQueues();
+
+  return {
+    queue: queues.find((q) => q.uuid == queueUuid),
```

Comment:
I'm curious about how @ibacher / @denniskigen / @brandones feel about this approach in terms of practices.  Basically, the argument that @chibongho is making is that there aren't a ton a queues in the system, and it will be common for users in the UI to toggle between different queues, so this minimizes network requests by fetching them all upfront and then finding what you need, rather than fetching them on demand.  It's an easy to change implementation detail, but would like to hear others thoughts.

Reply:
Does this just return the queues themselves, or their entries?  If it just a list of queue objects, this makes sense... if this returns all the active queue entries for all the queues, I guess it depends on how big we think that could grow, seems reasonable that it would never be *that* large.

Reply:
`useQueues() ` returns just the list of queue objects, not the queue entries. Given that we will have UI elements that also need to list out all possible queues (like the form to transfer patient to different queue), I think relying on `useQueues()` here will save requests.

Reply:
yeah, agreed 100%, thanks @chibongho

Reply:
Seems like the right approach to me!

Reply:
I don't have a problem with the approach per se, but:

> it will be common for users in the UI to toggle between different queues

This is absolutely not how I would expect _most_ users to use the OpenMRS at all. Most users should be looking at a single queue (basically, who do I need to see next); only someone like a clinic manager is likely to ever look at more than one queue.

Reply:
All great points.  Sounds like there are no major objections to the current approach, so I'm going to resolve this conversation in favor of leaving this alone.